custom_components/bosch_alarm/device.py

```diff
@@ -2,11 +2,36 @@
 
 from .const import DOMAIN
 
-def device_info_from_panel(panel):
+def device_info_from_panel(data):
+    panel = data.panel
     return DeviceInfo(
-        identifiers={(DOMAIN, panel.serial_number)},
-        name=f"Bosch {panel.model}",
+        identifiers={(DOMAIN, data.unique_id)},
+        name=f"Bosch {data.model}",
         manufacturer="Bosch Security Systems",
         model=panel.model,
         sw_version=panel.firmware_version,
     )
+
+class BoschPanel:
+    def __init__(self, panel, unique_id, model):
+        self.unique_id = unique_id
+        self.model = model
+        self.panel = panel
+        self._initialised = False
+        self._entity_setups = []
+        panel.connection_status_observer.attach(self.setup)
+
+    def register_entity_setup(self, setup):
+        self._entity_setups.append(setup)
+
+    def setup(self):
```

Comment:
This class doesn't actually know/care that it's dealing with setups.
Maybe call this on_connection_status_change, and rename _entity_setups to on_connect?